include/swift/IDE/Utils.h

```diff
@@ -543,6 +543,19 @@ class SourceEditOutputConsumer : public SourceEditConsumer {
   void accept(SourceManager &SM, RegionType RegionType, ArrayRef<Replacement> Replacements) override;
 };
 
+class DuplicatingSourceEditConsumer : public SourceEditConsumer {
+  SourceEditTextConsumer TextConsumer;
+  SourceEditOutputConsumer OutputConsumer;
+
+public:
+  DuplicatingSourceEditConsumer(llvm::raw_ostream &TextStream,
+                                SourceManager &SM, unsigned BufferId,
+                                llvm::raw_ostream &OutputStream);
```

Comment:
I think, `DuplicatingSourceEditConsumer` can be more generic by having two `SourceEditConsumer`s as members and having an initializer that just takes two `SourceEditConsumer`s. 

```suggestion
  DuplicatingSourceEditConsumer(SourceEditConsumer ConsumerA,
                                SourceEditConsumer ConsumerB);
```

Then, we could create a `SourceEditTextConsumer` and a `SourceEditOutputConsumer` in `swift-refactor.cpp`, create a `DuplicationgSourceEditConsumer` that dispatches to both and assign that to `pConsumer`. Does this make sense?